panther_power_control/src/relays_node.py

```diff
@@ -134,19 +149,50 @@ def _e_stop_reset_cb(self, req: TriggerRequest) -> TriggerResponse:
             return TriggerResponse(True, 'E-STOP reset successful')
 
     def _e_stop_trigger_cb(self, req: TriggerRequest) -> TriggerResponse:
-        with self._lock:
+        with self._e_stop_lock:
             if self._e_stop_state:
                 return TriggerResponse(True, 'E-SROP already triggered')
-            
+
             self._e_stop_state = True
             self._e_stop_state_pub.publish(self._e_stop_state)
             return TriggerResponse(True, 'E-SROP triggered successful')
 
+    def _motor_enable_cb(self, req: SetBoolRequest) -> SetBoolResponse:
+        with self._motors_lock:
+            if not self._lines['STAGE2_INPUT'].get_value():
+                self._motor_enabled = False
+                return SetBoolResponse(
+                    not req.data,
+                    f'Motors are {"already " if not req.data else ""}disabled. '
+                    + '(Main switch set to Stage 1)',
+                )
+
+            if self._motor_enabled == req.data:
+                return SetBoolResponse(
+                    True, f'Motors are already {"enabled" if self._motor_enabled else "disabled"}'
+                )
+
+            self._motor_enabled = req.data
+            self._lines['MOTOR_ON'].set_value(req.data)
+            self._publish_motor_state(req.data)
+            return SetBoolResponse(True, f'Motors {"enabled" if req.data else "disabled"}')
+
     def _set_motor_state_timer_cb(self, *args) -> None:
-        motor_state = self._lines['STAGE2_INPUT'].get_value()
-        self._lines['MOTOR_ON'].set_value(motor_state)
-        if self._io_state.motor_on != motor_state:
-            self._io_state.motor_on = motor_state
+        with self._motors_lock:
+            motor_state = self._lines['STAGE2_INPUT'].get_value()
+            # if switch changes from off to on overwrite service value
+            if not self._last_motor_state and motor_state:
```

Comment:
`self._lines['STAGE2_INPUT'].get_value()` bardzo często się pojawia w kodzie proponuje stworzyć lambde `is_stage2()`
```suggestion
            # if switch changes from off to on overwrite service value
            if not self._last_motor_state and self.is_stage2():
```

Reply:
It appears 4 times. If we had to add lambda, we would have to do this for both pins and also port it to `power_borad_node.py` to keep the same coding style